lib/core/notifications/components/notification-history.component.ts

```diff
@@ -47,43 +44,100 @@ export class NotificationHistoryComponent implements OnDestroy {
+    notifications: NotificationModel[] = [];
+    paginatedNotifications = [];
+    pagination: Pagination;
+
     constructor(
-        private notificationService: NotificationService, public storageService: StorageService) {
-        this.notifications = JSON.parse(storageService.getItem('notifications')) || [];
+        private notificationService: NotificationService,
+        public storageService: StorageService,
+        public cd: ChangeDetectorRef) {
+
+    }
+
+    ngOnInit() {
+        this.notifications = JSON.parse(this.storageService.getItem('notifications')) || [];
+    }
+
+    ngAfterViewInit(): void {
         this.notificationService.notifications$
-            .pipe(takeUntil(this.onDestroy$))
-            .subscribe((message) => {
-                this.notifications.push(message);
+        .pipe(takeUntil(this.onDestroy$))
+        .subscribe((notification: NotificationModel) => {
+            this.addNewNotification(notification);
+            this.cd.detectChanges();
+        });
+    }
+
+    ngOnDestroy() {
+        this.onDestroy$.next(true);
+        this.onDestroy$.complete();
+    }
 
-                if (this.notifications.length > this.MAX_NOTIFICATION_STACK_LENGTH) {
-                    this.notifications.shift();
-                }
+    addNewNotification(notification: NotificationModel) {
+        this.notifications.unshift(notification);
+
+        if (this.notifications.length > NotificationHistoryComponent.MAX_NOTIFICATION_STACK_LENGTH) {
+            this.notifications.shift();
+        }
 
-                storageService.setItem('notifications', JSON.stringify(this.notifications));
-            });
+        this.saveNotifications();
+        this.resetPagination();
+        this.updatePagination();
     }
 
-    isEmptyNotification(): boolean {
-        return (!this.notifications || this.notifications.length === 0);
+    saveNotifications() {
+        this.storageService.setItem('notifications', JSON.stringify(this.notifications.filter((notification) =>
+            notification.type !== NOTIFICATION_TYPE.RECURSIVE
+        )));
+    }
+
+    onMenuOpened() {
+        this.resetPagination();
+        this.updatePagination();
     }
 
     onKeyPress(event: KeyboardEvent) {
         this.closeUserModal(event);
     }
 
+    private closeUserModal($event: KeyboardEvent) {
+        if ($event.keyCode === 27) {
+            this.trigger.closeMenu();
+        }
+    }
+
     markAsRead() {
-        this.storageService.removeItem('notifications');
         this.notifications = [];
+        this.paginatedNotifications = [];
+        this.resetPagination();
+        this.storageService.removeItem('notifications');
     }
 
-    ngOnDestroy() {
-        this.onDestroy$.next(true);
-        this.onDestroy$.complete();
+    updatePagination() {
+        this.pagination.totalItems = this.notifications.length;
+        this.pagination.skipCount = this.pagination.maxItems + this.pagination.skipCount;
```

Comment:
don't get this assignment 
skipCount = maxItems + skipCount.
When you open it you call reset and update so the skip would be 5 + 0. why ?